### ISSUE TITLE
Improper error about else if collapse.

### ISSUE THREAD

**Issue by wdanilo** [rust-lang/rust-clippy#4971]
Hi, I've got the following code:

```rust
        if t1 > t2 {
            if      value >= t1 { ValueCheck::Correct }
            else if value >= t2 { ValueCheck::Warning }
            else                { ValueCheck::Error   }
        } else {
            if      value <= t1 { ValueCheck::Correct }
            else if value <= t2 { ValueCheck::Warning }
            else                { ValueCheck::Error   }
        }
```

Clippy tells me:

```
warning: this `else { if .. }` block can be collapsed
   --> monitor.rs:328:16
    |
328 |           } else {
    |  ________________^
329 | |             if      value <= t1 { ValueCheck::Correct }
330 | |             else if value <= t2 { ValueCheck::Warning }
331 | |             else                { ValueCheck::Error   }
332 | |         }
    | |_________^
```

I believe this should not happen.

**Comment by flip1995** [comment]
That is technically correct. You could write this as

```rust
if t1 > t2 {
    if      value >= t1 { ValueCheck::Correct }
    else if value >= t2 { ValueCheck::Warning }
    else                { ValueCheck::Error   }
} else if value <= t1 { 
    ValueCheck::Correct 
} else if value <= t2 { 
    ValueCheck::Warning 
} else { 
    ValueCheck::Error   
}
```

which has the same semantic. 

Anyway, I agree, that in this case not collapsing the `else { if` is more readable. Maybe we could bail out when the `ExprKind` of all `if` arms is an `if {..} else {}` expression and nothing else.

**Comment by Manish-Giri** [comment]
Hello, can I work on this?

**Comment by flip1995** [comment]
Sure! If you have any questions, just ask here or open a WIP PR anytime. A good place to start is this [documentation](https://github.com/rust-lang/rust-clippy/blob/master/doc/adding_lints.md) how clippy lints are implemented and tested.

**Comment by boxdot** [comment]
@Manish-Giri: Are you still working on this? Otherwise, I would like to try to fix it.

**Comment by Manish-Giri** [comment]
@boxdot Hello, sorry I thought this got re-assigned. I had already done some work on this earlier, so I'll get on it rightaway and finish it as soon as I can.

**Comment by rwardd** [comment]
Hi, since this has been inactive for a while, I thought I'd go ahead and implement a fix
